runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java

```diff
@@ -107,7 +107,7 @@ public void onConfigChanged(ConfigChanged event)
 		}
 	}
 
-	@Subscribe
+	@Subscribe(priority = 1) // run after all other onChatMessage plugins
```

Comment:
If this is the solution that we want to add, we should add a test to ensure this priority will always be higher than that of other `ChatMessage` subscribers.

Reply:
Not exactly sure how to go about writing this integration test. All other tests written for `ChatMessage` seem to be unit tests. Even the Emoji and Chat Color interaction is handled as a unit test rather than integration.

Seems to me the easiest way of checking the order would be to check the outcome of a "default" `ChatMessage` invocation (the one all subscribes to). I am not sure where that is located. That way we would know it is correct regardless of what other subscribers do as well as future proof for changes and new plugins. 

Personally I am in favour of my initial solution by using the information from `MessageNode` rather than `ChatMessage`. It is afterall the `MessageNode` that gets updated in all the plugins: Emoji, Chat Color and this. This would allow the plugin to change the icon without, indirectly, also changing the name.

Either way we should indeed have both unit and integration tests for this plugin.

Reply:
Adam said integration testing this might not be doable.

Showing it working with Color + Emoji + League:
![SS showing Chat Color + Emoji + League together](https://i.imgur.com/KzEEWoK.png)

Don't think any other plugins modifies the `ChatMessage` or `MessageNode` objects.